tests/h/search/query_test.py

```diff
@@ -89,12 +89,25 @@ def search(self, search, pyramid_request):
 
 
 class TestGroupFilter(object):
+    def test_matches_only_annotations_from_specified_group(self, search, Annotation, group):
+        Annotation(groupid='group2')
+        Annotation(groupid='group3')
+        group1_annotations = [Annotation(groupid=group.name).id,
+                              Annotation(groupid=group.name).id]
+
+        result = search.run({'group': group.name})
```

Comment:
Heh - you made the test consistent but it should be the _other_ way. `groupid` should be a _pubid_ (which is unique) not a name (which is not). The "group" argument in search queries is also a pubid.